src/ios/UsabillaCordova.swift

```diff
@@ -24,29 +23,27 @@ import Usabilla
                     self.masks = value as? [String]
                 } else if (key == "MASK_CHAR") {
                     self.maskChar = value as? String
+                } else if (key == "CUSTOM_VARS") {
+                    self.customVariables = value as? [String : Any]
                 } else {
-                    if (value is String) {
-                        arguments[key] = value as? String
-                    } else if (value is Bool) {
-                        arguments[key] = value as? Bool
-                    }                    
+                    debugPrint(key, value, separator: " -- ")
                 }
             }
         }
-        self.customVariables = arguments
     }
 
     // Iinitialize the SDK with your appId to target campaigns
     @objc(initialize:)
     func initialize(command: CDVInvokedUrlCommand) {
         self.command = command;
         extractCustomVariables(command: command)
-        Usabilla.customVariables = self.customVariables!
+        let newCustomVariables = self.customVariables!.mapValues { String(describing: $0) }
         Usabilla.initialize(
             appID: self.appId,
             completion: {
                 self.success(completed: true)
         })
+        Usabilla.customVariables = newCustomVariables
```

Comment:
Does this need to be split in definition and assignment? (row 40 and 46)

Reply:
@kioli this is because custom variable is should be called after initialise, there was an issue that custom variable is not getting into FE got fixed by this trick.